### DISCUSSION TITLE
UI Tests Failing?

### DISCUSSION THREAD

**Discussion by emmyxth** [CS342/.github#12] [Technical Questions] [answered]
### In what area do you have a technical challenge?
UI Testing
### Description

Hi! It appears that some of my UI tests are failing. Namely: 
<img width="948" alt="Screenshot 2566-01-23 at 14 50 11" src="https://user-images.githubusercontent.com/93239069/214169408-ce0b5b23-7304-4ecf-a9d9-36089478c8f8.png">
I'm unsure why this is the case, even though I basically copied over what was in the exemplar's UI tests. I've also tried adjusting the original timeout period from 0.5 to 2, to no avail. Could you tell me why this is happening?

### Reproduction

```
class EmmyThamakaison: XCTestCase {
    override func setUpWithError() throws {
        try super.setUpWithError()
        continueAfterFailure = false
    }
    
    func testEmmyThamakaisonandLinkedInLink() throws {
        let app = XCUIApplication()
        app.launch()
        
        app.collectionViews.buttons["Emmy Thamakaison"].tap()
        XCTAssert(app.staticTexts["Emmy Thamakaison"].waitForExistence(timeout: 2.0))
```
### Expected behavior

Passing the tests

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
@emmyxth  Thank you for posting your question!

Can you share a link to your current code state in a feature branch in the CS342 class application project? My suspicion is that there is not view displaying your name `"Emmy Thamakaison"` in the detail view that you have created. This is what line 
```swift
XCTAssert(app.staticTexts["Emmy Thamakaison"].waitForExistence(timeout: 2.0))
```
Checks: It looks for the static text in the app that is displaying "Emmy Thamakaison". Do you have some text like this in your own view?

**Reply by rachelkwu** [reply]
Hi @PSchmiedmayer, I'm having the exact same problem as Emmy, could you elaborate on your answer? To my knowledge, the tests that I initially created mimic the tests for Paul's page since I also added a Twitter functionality, and I just replaced Paul's name with my name, which is what I named the new page I created. I do not know why it is not passing a simple text check when the build clearly shows that my name and Twitter are being displayed correctly. Thank you!

**Reply by vishnuravi** [reply]
Hi @rachelkwu, I looked at Emmy's issue yesterday. The reason her test fails is that on the view she is testing, the text is formatted as "Emmy" and "THAMAKAISON" in separate text views aligned vertically in a VStack, whereas the test is looking for "Emmy Thamakaison" in a single text view. You might be running into the same problem. Check to make sure that the text on the view you are testing is formatted the same way as your test. If not, you can update your test.

Feel free to share the code for your view and your test here and we can help you debug it!

**Reply by PSchmiedmayer** [reply]
Hi @rachelkwu,

I just checked out your feature branch in this PR: https://github.com/CS342/CS342-2023-Application/pull/36 and it seems like your test are now passing.

As noted by @vishnuravi noted, @emmyxth had split up your name in two different text views. Therefore checking of the continuous string `"Emmy Thamakaison"` fails. In addition, @emmyxth have written your last name in all caps. 

If anyone else stumbles across this discussion: You can e.g., check that these two strings existing in a UI testing using
```
XCTAssert(app.staticTexts["Emmy"].waitForExistence(timeout: 2.0))
XCTAssert(app.staticTexts["THAMAKAISON"].waitForExistence(timeout: 2.0))
```

@emmyxth  As far as I can see, this would also resolved your question. Let us know if you have any follow-up questions. Otherwise feel free to mark the comment as an answer to other students can learn from the process to getting to a running UI test 👍
